### ISSUE TITLE
INDEX_SETTINGS does NOT override the default settings to create the ElasticSearch index

### ISSUE THREAD

**Issue by ngocm** [wagtail/wagtail#3612]
According to the Doc:
```
        'INDEX_SETTINGS': {
            'settings': {
                'index': {
                    'number_of_shards': 1,
                    'analysis': {
                        'analyzer': {
                            'default': {
                                'type': 'italian'
                            }
                        }
                    }
                }
```

In order for the settings of `analysis` to take affect:
```
        'INDEX_SETTINGS': {
            'settings': {
                'index': {
                    'number_of_shards': 1,
                 },
                 'analysis': {
                      'analyzer': {
                         'default': {
                          'type': 'italian'
                          }
                     }
                }
```

This is because of how the default `settings` is specified in `class ElasticsearchSearchBackend(BaseSearchBackend)`

**Comment by tokawpl** [comment]
Hey guys, 
also curious about the search_analyzer mapping.
Is it possible to change the search_analyzer via INDEX_SETTINGS?

I tried it, but still getting from GET _mappings the default mappings where the search_analyzer is set on standard.

Thanks!

**Comment by BertrandBordage** [comment]
It was just a documentation issue. I fixed it in #3943.
@ngocm, the second structure you put it the correct Elasticsearch configuration format, where `analysis` is not inside `index`. That’s why `INDEX_SETTINGS` was not working, Elasticsearch simply ignored what it considered as extra keys.

It could have been clearer if Elasticsearch complained about unrecognized arguments…
